lsp/client-extension/syntaxes/nickel.tmLanguage.json

```
@@ -0,0 +1,137 @@
+{
+  "$schema": "https://raw.githubusercontent.com/martinring/tmlanguage/master/tmlanguage.json",
+  "name": "Nickel",
+  "patterns": [
+    {
+      "include": "#keywords"
+    },
+    {
+      "include": "#strings"
+    },
+    {
+      "include": "#comments"
+    },
+    {
+      "include": "#operators"
+    },
+    {
+      "include": "#numbers"
+    },
+    {
+      "include": "#bools"
+    },
+    {
+      "include": "#storage"
+    },
+    {
+      "include": "#types"
+    },
+    {
+      "include": "#strings_interpolation"
+    }
+  ],
+  "repository": {
+    "storage": {
+      "patterns": [
+        {
+          "name": "storage.modifier",
+          "match": "\\b(let|in|forall|import|default|force|optional|priority|doc|rec)\\b"
+        },
+        {
+          "name": "storage.type",
+          "match": "\\b(fun)\\b"
+        }
+      ]
+    },
+    "strings_interpolation": {
+      "name": "string.interpolated",
+      "begin": "m%\"$",
+      "end": "\"%",
+      "patterns": [
+        {
+          "name": "constant.character.escape.interpolation.nickel",
+          "match": "''%$\\{"
+        },
+        {
+          "name": "constant.other.placeholder.nickel",
+          "begin": "%\\{",
+          "beginCaptures": {
+            "0": {
+              "name": "punctuation.section.curly.begin.nickel"
+            }
+          },
+          "end": "\\}",
+          "endCaptures": {
+            "0": {
+              "name": "punctuation.section.curly.end.nickel"
+            }
+          }
+        }
+      ]
+    },
+    "types": {
+      "patterns": [
+        {
+          "name": "support.class",
+          "match": "\\b(Number|Bool|Dyn|String|Array)\\b"
+        }
+      ]
+    },
+    "keywords": {
+      "patterns": [
+        {
+          "name": "keyword.control.nickel",
+          "match": "\\b(if|else|then|match)\\b"
+        }
+      ]
+    },
+    "strings": {
+      "name": "string.quoted.double.nickel",
+      "begin": "\"",
+      "end": "\"",
+      "patterns": [
+        {
+          "name": "constant.character.escape.nickel",
+          "match": "\\\\."
+        }
+      ]
+    },
+    "comments": {
+      "name": "comment.line.number-sign",
+      "patterns": [
+        {
+          "name": "comment.line.number-sign",
+          "match": "#(.+)"
+        }
+      ]
+    },
+    "numbers": {
+      "name": "constant.numeric",
+      "patterns": [
+        {
+          "name": "constant.numeric",
+          "match": "(\\d+)(\\.\\d+)"
+        }
+      ]
+    },
+    "bools": {
+      "name": "constant",
+      "patterns": [
+        {
+          "name": "constant",
+          "match": "\\b(true|false)\\b"
+        }
+      ]
+    },
```

Should we add the `null` value as well, here?

```suggestion
    },
    "null": {
      "name": "constant.null",
      "patterns": [
        {
          "name": "constant.null",
          "match": "\\b(null)\\b"
        }
      ]
    },
```